azure-pipelines/prep-release.yml

```diff
@@ -0,0 +1,39 @@
+trigger:
+  - none
+
+pr:
+  - none
+
+schedules:
+  - cron: "00 5 * * *"
+    displayName: "Prepare and tag release"
+    branches:
+      include:
+        - master
+    always: true
+
+jobs:
+  - job: TagRelease
+    displayName: 'Prepare mbed-tools release'
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '3.7'
+
+      - script: |
+          git config --global user.name "Monty Bot"
+          git config --global user.email "monty-bot@arm.com"
+
+          python -m pip install --upgrade tox
+
+          git checkout master
+          tox -e preprelease
+          git show
+          git push https://%GIT_TOKEN%@github.com/ARMmbed/mbed-tools.git master
+          git push https://%GIT_TOKEN%@github.com/ARMmbed/mbed-tools.git --tags
```

Comment:
We don't want to push the tag until the pre-release job is finished running, otherwise the pre-release job will see the tag after it finishes testing and will push a real release version.